src/uct/ib/ud/verbs/ud_verbs.c

```diff
@@ -137,7 +139,7 @@ uct_ud_verbs_ep_send_ctl(uct_ud_ep_t *ud_ep, uct_ud_send_skb_t *skb,
     } else {
         ucs_assert(!(flags & UCT_UD_IFACE_SEND_CTL_FLAG_INLINE));
     }
-    if (flags & UCT_UD_IFACE_SEND_CTL_FLAG_SOLICITED) {
+    if ((flags & UCT_UD_IFACE_SEND_CTL_FLAG_SOLICITED) && dev->has_cq_notify) {
```

Comment:
what's wrong if we send solicited if w/o event support? would it cause an error?

Reply:
Yes, `ibv_post_send()` will fail.

Reply:
ok

Reply:
@jgunthorpe - I'm just curious if this is standard verbs behavior. thanks !

Reply:
Standard? I'd say no, but it does look right, there is no trace of interrupt handling in EFA for CQs. Polling only.

Reply:
@jgunthorpe If I parse this correctly, you are saying that having CQ interrupt handling is not a strict requirement for verbs providers....

Reply:
According to IBTA spec, event notification (incl. solicited) is mandatory by an HCA:
![image](https://user-images.githubusercontent.com/2255631/108640841-9f3d8600-74a4-11eb-985f-12041e26445b.png)

Reply:
@yosefe EFA is not a IBTA compliant device. AFAIK they have never claimed IBTA or ROCE comparability. I do agree with your point so, if SW IBTA chapter defines the verb as mandatory, it is reasonable to assume that it is implemented and not optional.

Reply:
just to clarify - I think it's not so nice to detect that a mandatory verb is (not) supported by trying to use it, but it's not a blocker for this PR

Reply:
I see Gal asking questions that suggest they might implement it, maybe ask?

Reply:
@rajachan any comments on this?

Reply:
Pasha is correct; EFA is not an IBTA-compliant device.  When we support a feature, we definitely try to conform to the the IBTA spec as presented by the rdma-core interface, but there are "mandatory" sections we don't support.  While not a public commitment to support a feature, I'm positive we'll eventually have support for completion notifications.  I don't think we'll have that in the time frame NVIDIA is hoping for with this PR.